mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -119,6 +119,18 @@ final class EMLDAOptimizer extends LDAOptimizer {
       }
     }
 
+    // create term vertices for empty docs
+    val emptyDocTermVertices: RDD[(VertexId, TopicCounts)] = {
+      //empty documents
+      val emptyDocs = docs.filter { case (d: Long, tc: Vector) =>
+        tc.toBreeze.reduce(_+_) != BDV(0.0)
```

Comment:
Is this clearer to compute as a norm on `Vector` and just compare to 0 instead of going to breeze and reducing? also I think your case can be like `case (_, tc)`